apps/frontend/src/components/searchBar.tsx

```diff
@@ -0,0 +1,252 @@
+import {
+  Grid,
+  Button,
+  TextField,
+  Menu,
+  MenuItem,
+  useMediaQuery,
+} from "@mui/material";
+import martletPlaceLogo from "../images/martletplace-logo.png";
+import message from "../images/message.png";
+import filter from "../images/filter.png";
+import { useStyles } from "../styles/pageStyles";
+import { useState, ChangeEvent, useEffect, useCallback } from "react";
+import Filters from "./filters";
+import { useNavigate } from "react-router-dom";
+import * as React from "react";
+
+interface SearchObject {
+  query: string;
+  minPrice: number;
+  maxPrice: number;
+  status: string;
+  searchType: string;
+  latitude: number;
+  longitude: number;
+  sort: string;
+  page: number;
+  limit: number;
+}
+
+interface SearchBarProps {
+  onSearch: () => void;
+  sortBy: string;
+}
+
+const SearchBar: React.FC<SearchBarProps> = ({ onSearch, sortBy }) => {
+  const classes = useStyles();
+  const navigate = useNavigate();
+
+  const [showFilters, setShowFilters] = useState(false);
+  const [searchInput, setSearchInput] = useState("");
+  const [filters, setFilters] = useState<SearchObject>({
+    query: "",
+    minPrice: 0,
+    maxPrice: 100000,
+    status: "AVAILABLE",
+    searchType: "LISTING",
+    latitude: 0,
+    longitude: 0,
+    sort: "RELEVANCE",
+    page: 1,
+    limit: 20,
+  });
+
+  const toggleFilters = () => {
+    setShowFilters(!showFilters);
+  };
+
+  const handleSearchInputChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setSearchInput(event.target.value);
+  };
+
+  const handleFilterChange = (newFilters: Partial<SearchObject>) => {
+    setFilters((prevFilters) => ({ ...prevFilters, ...newFilters }));
+  };
+
+  const handleSearch = useCallback(() => {
+    const searchObject: SearchObject = {
+      ...filters,
+      query: searchInput,
+      sort: sortBy,
+    };
+    console.log("Search Object:", searchObject);
+    onSearch();
+    // API CALL HERE???
+  }, [filters, searchInput, sortBy, onSearch]);
+
+  useEffect(() => {
+    if (sortBy !== "") {
+      handleSearch();
+    }
+  }, [sortBy, handleSearch]);
+
+  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === "Enter") {
+      handleSearch();
+    }
+  };
+
+  const handleMessageRoute = () => {
+    navigate("/messages");
+  };
+
+  const handleAccountRoute = () => {
+    navigate("/user");
+  };
+
+  const handleReload = () => {
+    navigate("/");
+    window.location.reload();
+  };
+
+  const handleListingRoute = () => {
+    navigate("/listing/view");
+  };
+
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
```

Comment:
This could probably be more easily implemented with just a showMenu toggle variable and css, but since this is already here there's no huge need to change it

Reply:
I kept it the same since its not causing any issues